libs/sdk-backend-tiger/src/backend/workspaces/index.ts

```diff
@@ -53,41 +63,68 @@ class TigerWorkspaceQuery implements IWorkspacesQuery {
         return this.queryWorker(this.offset, this.limit, this.search);
     }
 
+    private resultToWorkspaceDescriptors = (result: JsonApiWorkspaceList): IWorkspaceDescriptor[] => {
+        return result.data.map(workspaceConverter);
+    };
+
+    private searchWorkspaceDescriptors = (search?: string) => (
+        results: IWorkspaceDescriptor[],
+    ): IWorkspaceDescriptor[] => {
+        if (search) {
+            const lowercaseSearch = search.toLocaleLowerCase();
+
+            return results.filter((workspace) => {
+                const { title } = workspace;
+
+                return title && title.toLowerCase().indexOf(lowercaseSearch) > -1;
+            });
+        }
+        return results;
+    };
+
+    private descriptorToAnalyticalWorkspace = (descriptor: IWorkspaceDescriptor): IAnalyticalWorkspace =>
+        new TigerWorkspace(this.authCall, descriptor.id, this.dateFormatter, descriptor);
+
+    private descriptorsToAnalyticalWorkspaces = (
+        descriptors: IWorkspaceDescriptor[],
+    ): IAnalyticalWorkspace[] => descriptors.map(this.descriptorToAnalyticalWorkspace);
+
     private async queryWorker(
         offset: number,
         limit: number,
         search?: string,
     ): Promise<IWorkspacesQueryResult> {
-        const emptyResult: IWorkspacesQueryResult = {
-            search,
-            items: [],
-            limit,
-            offset,
-            totalCount: 0,
-            next: () => Promise.resolve(emptyResult),
-        };
-        const workspaces = (
-            await this.authCall(async (sdk) => {
-                return sdk.organizationObjects.getAllEntitiesWorkspaces(
-                    {
-                        page: offset / limit,
-                        size: limit,
-                        sort: ["name"], // This is default behavior in gdc-bear
-                    },
-                    { headers: jsonApiHeaders },
-                );
-            })
-        ).data.data;
-        return {
-            search,
-            items: workspaces.map((workspace) => {
-                const descriptor = workspaceConverter(workspace);
-                return new TigerWorkspace(this.authCall, descriptor.id, this.dateFormatter, descriptor);
-            }),
-            limit,
-            offset,
-            totalCount: workspaces.length,
-            next: () => Promise.resolve(emptyResult),
+        const optionsToUse: OrganizationGetEntitiesOptions = {
+            query: {
+                size: limit,
+                page: offset,
+            },
```

Comment:
this is incorrect. page is different unit of measure than offset.

the query needs to start at page 0, load all, apply search, then start paging from the desired offset.

Reply:
I think you can omit these options completely. the utility will be adding its own paging..